bin/sql-formatter-cli.cjs

```diff
@@ -72,57 +73,73 @@ class SqlFormatterCli {
       process.exit(0);
     }
 
+    return {
+      language: this.args.language,
+      ...(await this.getConfig()),
+    };
+  }
+
+  async getConfig() {
     if (this.args.config) {
       // First, try to parse --config value as a JSON string
       try {
-        const configJson = JSON.parse(this.args.config);
-        return { language: this.args.language, ...configJson };
+        return JSON.parse(this.args.config);
       } catch (e) {
         // If that fails, try to read the --config value as a file
-        try {
-          const configFile = await this.readFile(this.args.config);
-          const configJson = JSON.parse(configFile);
-          return { language: this.args.language, ...configJson };
-        } catch (e) {
-          if (e instanceof SyntaxError) {
-            console.error(
-              `Error: unable to parse as JSON or treat as JSON file: ${this.args.config}`
-            );
-            process.exit(1);
-          }
-          this.exitWhenIOError(e);
-          console.error('An unknown error has occurred, please file a bug report at:');
-          console.log('https://github.com/sql-formatter-org/sql-formatter/issues\n');
-          throw e;
-        }
+        return this.parseFile(this.args.config);
       }
     }
-    return {
-      language: this.args.language,
-    };
+
+    // Otherwise find a local config file
+    const localConfig = await this.findConfig();
+    if (!localConfig) {
+      return null;
+    }
+
+    return this.parseFile(localConfig);
+  }
+
+  async findConfig(dir = __dirname) {
```

Comment:
It seems that this is not really an async function. At least the only `await` inside of it is calling the same function recursively.